rooms.js

```diff
@@ -163,6 +165,96 @@ var Room = (function () {
 		}
 		return true;
 	};
+	//mute handling
+	Room.prototype.runMuteTimer = function () {
+		if (this.muteTimer) return;
+		if (this.mutedUsers[0].time < Date.now() + 5) {
+			// this prevents lag from bugging the timer
+			this.mutedUsers[0].time = Date.now() + 5;
+		}
+		var self = this;
+		this.muteTimer = setTimeout(function () {
+			var user = Users(self.mutedUsers[0].userid);
+			if (user) {
+				user.popup("Your mute has expired.");
+				self.unmute(user);
+			}
+		}, self.mutedUsers[0].time - Date.now());
+	};
+	Room.prototype.isMuted = function (user) {
+		if (!user) return;
+		if (this.mutedUsers) {
+			for (var i = 0; i < this.mutedUsers.length; i++) {
+				if (user.userid === this.mutedUsers[i].userid ||
+					user.guestNum === this.mutedUsers[i].guestNum ||
+					(user.autoconfirmed && user.autoconfirmed === this.mutedUsers[i].autoconfirmed)) {
+					return this.mutedUsers[i].userid;
+				}
+			}
+		}
+	};
+	Room.prototype.getMuteTime = function (user) {
+		var userid = this.isMuted(user);
+		if (!userid) return;
+		for (var i = 0; i < this.mutedUsers.length; i++) {
+			if (userid === this.mutedUsers[i].userid) {
+				return this.mutedUsers[i].time - Date.now();
+			}
+		}
+	};
+	Room.prototype.mute = function (user, setTime) {
+		var userid = user.userid;
+
+		if (!setTime) setTime = 7 * 60000; // default time: 7 minutes
+		if (setTime < 1) setTime = 1; // mostly to prevent bugs
+		if (setTime > 90 * 60000) setTime = 90 * 60000; // limit 90 minutes
+
+		// Place the user in a queue for the unmute timer
+		for (var queue = 0; queue < this.mutedUsers.length + 1; queue++) {
+			var timeRemaining = this.mutedUsers[queue] ? this.mutedUsers[queue].time - Date.now() : Infinity;
+			if (setTime < timeRemaining) {
+				var buffer = Object.create(null);
+				buffer.userid = userid;
+				buffer.time = Date.now() + setTime;
+				buffer.guestNum = user.guestNum;
+				buffer.autoconfirmed = user.autoconfirmed;
+				this.mutedUsers.splice(queue, 0, buffer);
+				break;
+			}
+		}
+		this.runMuteTimer();
+
+		user.updateIdentity(this.id);
+
+		return userid;
+	};
+	Room.prototype.unmute = function (user, noRecurse) {
+		var userid = user.userid;
+		var successUserid = false;
+		for (var i = 0; i < this.mutedUsers.length; i++) {
+			var entry = this.mutedUsers[i];
+			if (entry.userid === userid ||
+				entry.guestNum === user.guestNum ||
+				(entry.autoconfirmed && entry.autoconfirmed === user.autoconfirmed)) {
+				if (i === 0) {
+					clearTimeout(this.muteTimer);
```

Comment:
I'd guess when you had me move to checking for existing muteTimer in `runMuteTimer()` I only tested muting and unmuting a single user, which correctly set muteTimer to `null` with how it was structured previously, and just assumed that I hadn't broken the function from the one line move.

Poor excuses aside, I tested everything again more thoroughly and everything seems to be working as far as I can tell.